### PR TITLE
Fix  #1674: Custom MyCodes not accepted by editor.

### LINKED ISSUE #1674: Custom MyCodes not accepted by editor.
> Create a new custom MyCode:
> Regular Expression:
> 
> ``` regex
> \[code=gist\](.*?)\[/code\]
> ```
> 
> Now try your new MyCode in a new thread/post. The code gets replaced by a common `[code]` MyCode.
> 
> Full or source editor doesn't makes the difference it seems.

Original thread: [Custom MyCodes not accepted by editor.](http://community.mybb.com/thread-163255.html)

### PR DESCRIPTION
**Pull request by martec**
Fix  #1674

### CODE REVIEW

**Comment by Sama34** [comment]
I honestly doubt this will make it to 1.8.5 but can you explain what your change does?

**Comment by martec** [comment]
@Sama34 
changes made in this file https://github.com/samclarke/SCEditor/blob/master/src/plugins/bbcode.js
after merged with https://github.com/mybb/mybb/blob/feature/jscripts/sceditor/jquery.sceditor.min.js
and replaced with jquery.sceditor.bbcode.min.js

and removed duplicated tags from bbcodes_sceditor.js

**Comment by Sama34** [comment]
I will take a look later but I really doubt this makes it to 1.8.5

**Comment by martec** [comment]
no problem if it is to 1.8.6

**Comment by Destroy666x** [comment]
So this is still the same solution (modify all BBCodes) without duplicating the code?

**Comment by martec** [comment]
@Destroy666x 
Yes, exactly.

File: install/resources/mybb_theme.xml [hunk 1]
```diff
@@ -13793,7 +13793,7 @@ if(use_xmlhttprequest == "1")
 </tr>]]></template>
 		<template name="codebuttons" version="1805"><![CDATA[<link rel="stylesheet" href="{$mybb->asset_url}/jscripts/sceditor/editor_themes/{$theme['editortheme']}" type="text/css" media="all" />
 <script type="text/javascript" src="{$mybb->asset_url}/jscripts/sceditor/jquery.sceditor.bbcode.min.js?ver=1805"></script>
-<script type="text/javascript" src="{$mybb->asset_url}/jscripts/bbcodes_sceditor.js?ver=1804"></script>
+<script type="text/javascript" src="{$mybb->asset_url}/jscripts/bbcodes_sceditor.js?ver=1805"></script>
```
Comment (JN-Jones) [inline install/resources/mybb_theme.xml]: Should be `1806` now. Also the template version needs to be bumped.

**Comment by martec** [comment]
@JN-Jones 
done

**Comment by JN-Jones** [comment]
Produced conflicts with one of the other PRs @martec

**Comment by Destroy666x** [comment]
Note: the bbcode plugin file should be minified like it was before.

Also, keep in mind that merging this will most likely make SCEditor much harder.

**Comment by martec** [comment]
Then better to first decide whether to use this solution or look for another. Because solve concflict issue is pretty boring.

**Comment by Sama34** [comment]
+1

**Comment by Destroy666x** [comment]
We should probably at least generate some kind of a readable diff or something for easier comparison at a later stage.

**Comment by martec** [comment]
i found one problem in this...

not possible set if will become block level element or inline element...

explain better....
if use something this [code=xxx][/code] will become block level element... because code is block level element.
 if use something this [b=xxx][/b] will become inline element... because bold is inline element...

same analogy of div element (block element) and span (inline element)

**Comment by dequeues** [comment]
This PR has merge conflicts that need to be resolved.

**Comment by martec** [comment]
not good idea use this...
so discard it.
better recreate all tags that mybb using even causing increase file size.

**Comment by dequeues** [comment]
Closing this then.

**Pull request closed without merging**